src/states/elder/mod.rs

```diff
@@ -857,9 +841,15 @@ impl Elder {
         self.vote_for_network_event(event.into_network_event())
     }
 
-    fn vote_for_section_info(&mut self, info: EldersInfo) -> Result<(), RoutingError> {
-        let signature_payload = SectionInfoSigPayload::new(&info, &self.full_id)?;
-        self.vote_for_network_event(info.into_network_event_with(Some(signature_payload)));
+    fn vote_for_signed_event<T: IntoAccumulatingEvent + Serialize>(
```

Comment:
Possibly worth having 2 named functions for the different events, it does help with searching for when things are voted, particularly the section_info one.